File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<int> h = new HashSet<int>();
        int slow = 0;
        int max = 0;

        for(int i=0;i<s.Length;i++){
            char c = s[i];
            if(h.Contains(c)){
                while(s[slow] != c){
                    h.Remove(s[slow]);
                    slow++;
                }
                slow++;
            }else{
                h.Add(c);
            }
            max = Math.Max(max,i-slow +1);
        }

    return max;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        int r = 0;
        int l = 0;

        unordered_map<int,int> map;
            int maxval = 0;
        for(int r = 0; r<s.size();r++){
            if(map.find(s[r]) == map.end()){
                map[s[r]] = 1;
            }else{
                while(map.find(s[r]) !=map.end() && l<r){
                    map.erase(s[l]);
                    l++;
                }
                map[s[r]] = 1;
            }
            maxval = max(maxval,(r-l+1));
        }

        return maxval;
    }
};